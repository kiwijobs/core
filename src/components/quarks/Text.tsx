import React, { FC } from 'react';
import { Box, BoxProps } from './Reflexbox';

export const Text: FC<BoxProps> = props => <Box as="span" {...props} />;
