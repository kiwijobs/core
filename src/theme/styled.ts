export {
  default as styled,
  css,
  keyframes,
  createGlobalStyle,
  StyledComponent,
  ThemeProvider,
  DefaultTheme,
} from 'styled-components';
import 'styled-components';
import { theme } from './theme';

declare module 'styled-components' {
  type ITheme = typeof theme;
  export interface DefaultTheme extends ITheme {}
}
