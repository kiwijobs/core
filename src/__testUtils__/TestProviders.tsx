import React, { FC, Suspense } from 'react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../theme';

export const TestProviders: FC<{}> = ({ children }) => {
  return (
    <ThemeProvider theme={theme}>
      <Suspense fallback="mocked loading...">{children}</Suspense>
    </ThemeProvider>
  );
};

export const wrapper = TestProviders; // React Testing Library friendly name.
