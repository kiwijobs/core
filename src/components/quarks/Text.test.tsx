import React from 'react';
import styled from 'styled-components';
import { Text } from './Text';
import { render } from '@testing-library/react';
import { wrapper } from '../../__testUtils__';

describe('Text', () => {
  it('Has "fontScale" prop', () => {
    const container = render(<Text fontScale={2} />, { wrapper }).container.firstChild;
    expect(container).toHaveStyleRule('font', `1.2rem / 1.6rem "Open Sans"`);
  });

  it('Overwrites fontScale', () => {
    const container = render(<Text fontScale={5} fontSize="33px" />, { wrapper }).container
      .firstChild;
    expect(container).toHaveStyleRule('font-size', '33px');
    expect(container).toHaveStyleRule('font', `1.8rem / 2.4rem "Open Sans"`);
  });

  it('Works with styled-components', () => {
    const Button = (props: any) => (
      <Text
        sx={{
          color: 'bahamaYellow',
          padding: '15px 30px',
          opacity: 1,
        }}
        fontScale={5}
        {...props}
      />
    );

    const StyledButtonLink = styled(Button)`
      padding-right: 45px;
      z-index: 2;
    `;
    const ReflexButtonLink = (props: any) => (
      <Button
        css={`
          padding-right: 60px;
          margin: 5px;
          z-index: 5;
        `}
        fontScale={6}
        {...props}
      />
    );

    expect(render(<StyledButtonLink />, { wrapper }).container).toMatchSnapshot();
    expect(render(<StyledButtonLink margin="20px 0" />, { wrapper }).container).toMatchSnapshot();
    expect(render(<ReflexButtonLink margin="20px 0" />, { wrapper }).container).toMatchSnapshot();
  });
});
