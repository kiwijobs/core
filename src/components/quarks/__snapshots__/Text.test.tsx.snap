// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Works with styled-components 1`] = `
.c0 {
  color: rgba(255,231,157,1);
  padding: 15px 30px;
  opacity: 1;
  font: 1.8rem / 2.4rem "Open Sans";
}

.c1 {
  padding-right: 45px;
  z-index: 2;
}

<div>
  <span
    class="c0 c1"
  />
</div>
`;

exports[`Text Works with styled-components 2`] = `
.c0 {
  color: rgba(255,231,157,1);
  padding: 15px 30px;
  opacity: 1;
  font: 1.8rem / 2.4rem "Open Sans";
  margin: 20px 0;
}

.c1 {
  padding-right: 45px;
  z-index: 2;
}

<div>
  <span
    class="c0 c1"
  />
</div>
`;

exports[`Text Works with styled-components 3`] = `
.c0 {
  color: rgba(255,231,157,1);
  padding: 15px 30px;
  opacity: 1;
  padding-right: 60px;
  margin: 5px;
  z-index: 5;
  font: 2rem / 2.4rem "Open Sans";
  margin: 20px 0;
}

<div>
  <span
    class="c0"
  />
</div>
`;
